### ISSUE TITLE
Gemini real-time

### ISSUE THREAD

**Issue by Shandelier** [livekit/agents#1217]
Any plans for implementing Gemini real-time API to Agents?

**Comment by jadams777** [comment]
+1 for this

**Comment by CrackheadCode** [comment]
+1

**Comment by davidzhao** [comment]
we are working on it. stay tuned!

**Comment by pixelcatgg** [comment]
+1 and thank you @davidzhao looking forward to it

**Comment by mpisat** [comment]
If everyone enters their own api key for demo, it should work fine. Just like openai key requirement at the demo site.

**Comment by jadams777** [comment]
Which demo?

**Comment by mpisat** [comment]
> Which demo?

https://playground.livekit.io/
which is outdated I think?

**Comment by OrkhanGG** [comment]
> +1 and thank you @davidzhao looking forward to it. something interesting to know is it seems they have a limit of
> 
> ```
> 3 concurrent sessions per API key
> ```
> 
> is there guidance of how to queue up or rotate api keys? the rates seem pretty low for an actual application

There might be some workarounds to rotate the keys and bypass API usage limits, but I’m not sure if that complies with Google's policies. Since this is an experimental model, I strongly recommend against using it in production for now. The rates will likely increase once the stable version is released.

**Comment by smxthereisonlyone** [comment]
Any update here and when it will support video input?

**Comment by arthurblake** [comment]
Gemini 2 looks EXTREMELY impressive, and our early tests are showing that it's way smarter than OpenAI in our particular business domain. I've been asked to shift over to Gemini MultiModal for my livekit development. So I'mreally looking forward to this support...

**Comment by sajithlaldev** [comment]
+1, looking forward to it

**Comment by irux** [comment]
@davidzhao any PR for this ?

**Comment by omer-dtc** [comment]
+1

**Comment by arthurblake** [comment]
This came out with Agents for Python 0.12.6 Release on New Years Eve. The demo is here: 
https://github.com/livekit/agents/blob/main/examples/multimodal-agent/gemini_agent.py
I got it to work, although I am having trouble figuring out some basic things, like how to send an initial message from the agent when it joins the room

**Comment by jadams777** [comment]
@arthurblake  Any chance you could share with us how to get the demo app running?

**Comment by arthurblake** [comment]
> @arthurblake Any chance you could share with us how to get the demo app running?

Nothing special, it worked as is. If you are having specific problems you should ask on the slack channel.

**Comment by omer-dtc** [comment]
@arthurblake yes but it's missing support for video streams.

**Comment by snipd-min** [comment]
@arthurblake did you notice that the agent produces much less logs somehow compared to the OpenAI realtime agent, e.g. when calling functions?

**Comment by arthurblake** [comment]
Yes seems like they rushed it out. I did not try to use video because my main use case is for sip calls / audio only. Hopefully subsequent versions will have these issues fixed. Best to discuss on slack.

**Comment by AmbujaAK** [comment]
I'm facing some issues while running the examples of Gemini in real-time. Do you know if anyone has some working examples that can be shared?
